src/sink/wavefront.rs

```diff
@@ -247,7 +251,13 @@ impl Wavefront {
 
         let mut tag_buf = String::with_capacity(1_024);
         let aggrs = mem::replace(&mut self.aggrs, Buckets::default());
-        for value in padding(aggrs.into_iter().filter(|x| x.is_zeroed()), self.bin_width) {
+        for value in
+            padding(aggrs.into_iter().filter(|x| x.is_zeroed()), self.bin_width)
+        {
+            if value.persist {
+                let new_val = value.clone();
+                self.aggrs.add(new_val.timestamp(value.timestamp + 1));
```

Comment:
It is. `self.aggrs` is different from `aggrs` which got consumed previously. When we do `mem::replace` above what we're doing is allocating a new `Bucket`, sticking it into the reference at `self.aggrs` and returning `aggrs`. 

That is, `self.aggrs` is a pointer into the `Self` == `Wavefront` struct and `aggrs` is a locally scoped heap object.